pkgs/test/README.md

```diff
@@ -518,6 +521,37 @@ the following HTML file:
 </html>
 ```
 
+### Providing a custom HTML template
+
+If you want to share the same HTML file across all tests, you can provide a
+`custom-html-template-path` configuration option to your configuration file.
+This file should follow the rules above, except that instead of the link tag
+add exactly one `{testScript}` in the place where you want the template processor to insert it.
+
+The template can't end with `.html` since the test loader will not able to process it and an error will be thrown.
```

Comment:
I am a bit confused by why it errors - does it glob for all html files at some point? I wouldn't expect the test loader to ever try to load this file.

Reply:
Sorry, this might be my bad writing. It will throw because I made it so in https://github.com/dart-lang/test/pull/1127/files/343af3e063c748e1323177e82854dbb0aaf1a33f#diff-b9a58c9d38842cb742b063d60ff6fb18R257 But the point of this is that if it is an `.html` file it will be served directly by shelf and will not be handled by the wrapper in `platform.dart`

Reply:
This file should only ever need to be read from disk as a part of the `platform.dart` code though (it doesn't go through the shelf handler) - and it should be fine for shelf to serve it directly even if it was requested for some reason?

Reply:
Hmm, so the issue I was having is that the `suiteUrl` is generated in the `load` method by taking the dart file name and replacing `.dart` with `.html`. If shelf then tries to load that html file but it doesn't exist on disk, it will go into the wrapper handler and generate the file.
However if it does, it will be served directly, so it is not possible to process it, so it can't really be used as a template anymore. I came to the conclusion that it's less confusing if the template can't be an html file, because then it would be hard to understand without looking at the internal mechanics why it isn't processed. I decided then to disallow that during the config parsing phase.

Sure, it is kind of an edge case since the template file would have to be named like the test file, but I think it's nice to separate the old mechanics of having a custom html file for the test and having a template file. But if you think it should be allowed, then I'll remove that check.

Reply:
Ah ok I get where the error came from now - IMO I don't think that is a common enough case to block all `.html` file extensions though, cc @natebosch wdyt?

Reply:
+1. If anything we should check that it doesn't match exactly the `.html` file for an existing test case.

Reply:
Ah ya we could change the check to that instead - I like that.